vyper/parser/context.py

```diff
@@ -146,8 +146,8 @@ def is_valid_varname(self, name, pos):
 
     # TODO location info for errors
     # Add a new variable
-    def new_variable(self, name, typ, pos=None):
-        if self.is_valid_varname(name, pos):
+    def new_variable(self, name, typ, anon=False, pos=None):
```

Comment:
Should we rename `anon` to be something more descriptive?  For example, maybe we could have `check_varname=True` instead.  It would also be nice to add a note about why that option exists and why it's useful.

Reply:
I'm also seeing a couple of other places (like [here](https://github.com/ethereum/vyper/blob/ab39d4e9c5168eff2646dd29e23d7212654b636d/vyper/parser/events.py#L227)) where internal variables are being created without the `new_placeholder` method.  So it might actually make more sense to have a new method, `new_internal_variable` which ensures that the provided variable name doesn't collide with the namespace in the vyper source file.  Then `new_placeholder` could just make a call to that without manually mangling its variable name.

Reply:
Good catch and suggestion. I'm thinking of just putting the mangling option into new_variable instead of a whole new function.

Reply:
@davesque let me know what you think of be9fc1d11b749c21165d72d7ae82375199a08f68